### PR TITLE
 Add encryption key option in workflow to test and deploy

### PR DESCRIPTION
**Pull request by htoo-eain-lwin**
## What happened 👀

- To run Munit with CD, we need to run `mvn test` with `securedKey` as an ENV.
- To overwrite properties in pom.xml for deployment, we need to add `-DencryptionKey` when deploy

## Insight 📝

use ENCRYPTION_KEY as the secrets for both `.github/workflows/shared_test.yml` and `.github/workflows/shared_deploy_cloudhub_1_0.yml`

### CODE REVIEW

File: .github/workflows/shared_test.yml [hunk 1]
```diff
@@ -17,6 +17,9 @@ on:
         required: true
       NEXUS_PASSWORD:
         required: true
+      ENCRYPTION_KEY_DEV:
+        required: true
+
```
Comment (andyduong1920) [inline .github/workflows/shared_test.yml:22]:
```suggestion
```

File: .github/workflows/shared_deploy_cloudhub_1_0.yml [hunk 2]
```diff
@@ -21,6 +21,8 @@ on:
         required: true
       CLOUDHUB_BUSINESS_GROUP_ID:
         required: true
+      ENCRYPTION_KEY_DEV:
```
Comment (andyduong1920) [inline .github/workflows/shared_deploy_cloudhub_1_0.yml]:
```suggestion
      ENCRYPTION_KEY:
```

File: .github/workflows/shared_deploy_cloudhub_1_0.yml [hunk 3]
```diff
@@ -51,3 +53,4 @@ jobs:
           use_artifact: false
           mule_file_path: ${{ steps.build.outputs.mule_file_path }}
           mule_runtime_version: ${{ inputs.mule_runtime_version }}
+          encryption_key: ${{ secrets.ENCRYPTION_KEY_DEV }}
```
Comment (andyduong1920) [inline .github/workflows/shared_deploy_cloudhub_1_0.yml]:
```suggestion
          encryption_key: ${{ secrets.ENCRYPTION_KEY }}
```

File: deploy_cloudhub_1_0/action.yml [hunk 4]
```diff
@@ -44,7 +47,7 @@ runs:
       shell: bash
       run: |
         artifactName=${{ inputs.mule_file_path }}
-        mvn deploy -DskipTests -DmuleDeploy -Dartifact=${artifactName}
+        mvn deploy -DskipTests -DmuleDeploy -Dartifact=${artifactName} -DencryptionKey=${ENCRYPTION_KEY_DEV}
```
Comment (andyduong1920) [inline deploy_cloudhub_1_0/action.yml]:
```suggestion
        mvn deploy -DskipTests -DmuleDeploy -Dartifact=${artifactName} -DencryptionKey=${ENCRYPTION_KEY}
```

File: deploy_cloudhub_1_0/action.yml [hunk 5]
```diff
@@ -53,3 +56,4 @@ runs:
         CLOUDHUB_REGION: ${{ inputs.cloudhub_region }}
         CLOUDHUB_APPLICATION_NAME: ${{ inputs.cloudhub_application_name }}
         MULE_RUNTIME_VERSION: ${{ inputs.mule_runtime_version }}
+        ENCRYPTION_KEY_DEV: ${{ inputs.encryption_key }}
```
Comment (andyduong1920) [inline deploy_cloudhub_1_0/action.yml]:
```suggestion
        ENCRYPTION_KEY: ${{ inputs.encryption_key }}
```

**Review by andyduong1920 — COMMENTED** [review]
We would need to adjust the REAME as well @htoo-eain-lwin

File: .github/workflows/shared_test.yml (hunk 1 above)
Comment (htoo-eain-lwin) [reply .github/workflows/shared_test.yml:22]: Solved in b3e1c006ace2fbbe6c0c90ba6c8a002d8b9cca21

**Review by andyduong1920 — APPROVED** [review]

**Review by tednguyendev — APPROVED** [review]

**Pull request merged**